src/generator/generator.ts

```diff
@@ -59,12 +61,22 @@ export async function protocolGen(host: Host) {
     if (pagers.length > 0) {
       host.WriteFile(`${filePrefix}pagers.go`, pagers, undefined, 'source-file-go');
     }
-    const pollers = await generatePollers(session);
-    if (pollers.length > 0) {
-      const pollingHelper = await generatePollersHelper(session);
-      host.WriteFile(`${filePrefix}pollers_helper.go`, pollingHelper, undefined, 'source-file-go');
-      host.WriteFile(`${filePrefix}pollers.go`, pollers, undefined, 'source-file-go');
+    if (openapiType === 'data-plane') {
```

Comment:
Can this logic be pushed into `generatePollers()`, and have it return the correct content?

Reply:
So I personally think it's better to keep arm poller code separate from data-plane poller code and would like to keep them in two separate places. I can add this logic in generatePollers and then call the corresponding func from there.

Reply:
done

Reply:
Agreed we should keep them separate.  But generator.ts shouldn't need to know anything about ARM/data-plane, it just calls some generic `createPollers()` func and the correct content is returned.